### PR TITLE
Ltac2: allow 0 argument externals

### PR DESCRIPTION
**Pull request by SkySkimmer**
This will let us have non-thunk Array.empty and a couple others.

Overlays:
- https://github.com/mit-plv/rewriter/pull/102

### CODE REVIEW

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Comment by ppedrot** [comment]
The CI failures are legit, it seems.

**Comment by SkySkimmer** [comment]
rewriter needs ml overlay
bedrock needs .v overlay

What's our policy about ltac2 compatibility?

**Comment by JasonGross** [comment]
> rewriter needs ml overlay bedrock needs .v overlay
> 
> What's our policy about ltac2 compatibility?

Rewriter should have version-specific ML files, IIRC, so it shouldn't be too hard to make an overlay that doesn't break the build for previous versions

**Comment by ppedrot** [comment]
@SkySkimmer can you split this PR in two, one allowing 0-argument externals, and the other performing the changes? That will help making progress.

**Comment by SkySkimmer** [comment]
@coqbot run full ci

**Comment by SkySkimmer** [comment]
Moved non compatible changes to https://github.com/coq/coq/pull/17534

**Comment by SkySkimmer** [comment]
@coqbot run full ci

File: plugins/ltac2/tac2quote.ml [hunk 1]
```diff
@@ -546,5 +546,5 @@ let of_format { v = fmt; loc } =
     with Tac2print.InvalidFormat ->
       CErrors.user_err ?loc (str "Invalid format")
   in
-  let stop = CAst.make @@ CTacApp (global_ref (kername format_prefix "stop"), [of_tuple []]) in
+  let stop = global_ref (kername format_prefix "stop") in
```
Comment (ppedrot) [inline plugins/ltac2/tac2quote.ml:549]: You need to revert this line for this PR.

**Comment by ppedrot** [comment]
@SkySkimmer the overlay must be tweaked after #17533.

**Comment by SkySkimmer** [comment]
https://github.com/mit-plv/rewriter/pull/101 is not yet merged upstream

**Comment by ppedrot** [comment]
Indeed, but the overlay will need a fix nonetheless. Maybe you should rebase mit-plv/rewriter/pull/102 over mit-plv/rewriter/pull/101.

**Comment by SkySkimmer** [comment]
Overlay didn't need tweaking

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**